apps/renderscript/jni/rstest.cpp

```diff
@@ -0,0 +1,232 @@
+                std::cout.width(2);
+
+                char buffer[33];
+                uint8_t actual_value =
+                    actual.host[i * actual.stride[0] + j * actual.stride[1] +
+                                k * actual.stride[2]];
+                uint8_t expected_value =
+                    expected
+                        .host[i * expected.stride[0] + j * expected.stride[1] +
+                              k * expected.stride[2]];
+                if (actual_value != expected_value) {
+                    sprintf(buffer, "%2d(%2d)", actual_value, expected_value);
+                } else {
+                    sprintf(buffer, "%2d", actual_value);
+                }
+                std::cout << buffer;
+            }
+            std::cout << "]";
+        }
+
+        std::cout << std::endl;
+    }
+
+    return count_mismatches == 0;
+}
+
+buffer_t make_planar_image(int width, int height, int channels,
+                           uint8_t host[]) {
+    buffer_t bt_input = { 0 };
+    bt_input.host = &host[0];
+    bt_input.host_dirty = true;
+    bt_input.stride[0] = 1;
+    bt_input.extent[0] = width;
+    bt_input.stride[1] = width;
+    bt_input.extent[1] = height;
+    bt_input.stride[2] = width * height;
+    bt_input.extent[2] = channels;
+    bt_input.elem_size = 1;
+    return bt_input;
+}
+
+buffer_t make_interleaved_image(int width, int height, int channels,
+                                uint8_t host[]) {
+    buffer_t bt_input = { 0 };
+    bt_input.host = &host[0];
+    bt_input.host_dirty = true;
+    bt_input.stride[0] = 4;
+    bt_input.extent[0] = width;
+    bt_input.stride[1] = 4 * width;
+    bt_input.extent[1] = height;
+    bt_input.stride[2] = 1;
+    bt_input.extent[2] = channels;
+    bt_input.elem_size = 1;
+    return bt_input;
+}
+
+bool test(buffer_t bt_input, buffer_t bt_output, buffer_t bt_output_arm,
+          int(generated_rs)(buffer_t *_input_buffer, buffer_t *_result_buffer),
+          int(generated_arm)(buffer_t *_input_buffer,
+                             buffer_t *_result_buffer)) {
+    for (int j = 0; j < std::min(bt_input.extent[1], 10); j++) {
+        for (int i = 0; i < std::min(bt_input.extent[0], 10); i++) {
+            std::cout << " [";
+            for (int k = 0; k < bt_input.extent[2]; k++) {
+                std::cout.width(2);
+
+                char buffer[33];
+                sprintf(buffer, "%d", bt_input.host[i * bt_input.stride[0] +
+                                                    j * bt_input.stride[1] +
+                                                    k * bt_input.stride[2]]);
+                std::cout << buffer;
+            }
+            std::cout << "]";
+        }
+
+        std::cout << std::endl;
+    }
+    timeval t1, t2;
+
+    const int n_reps = 500;
+    gettimeofday(&t1, NULL);
+    int error = 0;
+    for (int i = 0; i < n_reps; i++) {
+        error = generated_rs(&bt_input, &bt_output);
```

Comment:
Yeah, @zvookin raised similar concern. In c714d82 I display best and worst time as well as iterations they were produced on:
```
Interleaved(vectorized) blur:
...
Out of 500 runs best times are:
RS:  1.607000ms(@142) 
ARM: 1.481000ms(@178)
Out of 500 runs worst times are:
RS:  428.874000ms(@0)
ARM: 2.150000ms(@0)
```